src/rebar_fetch.erl

```diff
@@ -61,7 +61,7 @@ download_source(AppDir, Source, State) ->
     catch
         C:T ->
             ?DEBUG("rebar_fetch exception ~p ~p ~p", [C, T, erlang:get_stacktrace()]),
-            throw(?PRV_ERROR({fetch_fail, Source}))
+            ?PRV_ERROR({fetch_fail, Source})
```

Comment:
There's an interesting difference in that we match `{error, Reason}` and throw `Reason` at the call site. `?PRV_ERROR` as a macro is `{error, {?MODULE, Reason}`. If we want to keep matching the semantics, we have to change the call sites to `throw({error, Reason})` to keep it compatible.

Otherwise we'll change an error we handle into an uncontrolled crash.